org.mixedrealitytoolkit.uxcore/Button/StatefulInteractableColliderToggle.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+
+using UnityEngine;
+
+namespace MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// The <see cref="StatefulInteractableColliderToggle"/> class is responsible for managing the state of the collider 
+    /// associated with a <see cref="StatefulInteractable"/>.
+    /// </summary>
+    /// <remarks>
+    /// This class will enables or disables the colliders associated with <see cref="StatefulInteractable"/> based on
+    /// the enabled state of <see cref="StatefulInteractable"/>.
+    /// </summary>
+    [AddComponentMenu("MRTK/UX/Stateful Interactable Collider Toggle")]
+    public class StatefulInteractableColliderToggle : MonoBehaviour
+    {       
+        [SerializeField]
+        [Tooltip("The StatefulInteractable to enable or disable the collider based on the interactable's enabled state.")]
+        private StatefulInteractable statefulInteractable;
+
+        /// <summary>
+        /// The <see cref="StatefulInteractable"/> to enable or disable the collider based on the interactable's enabled state.
+        /// </summary>
+        public StatefulInteractable StatefulInteractable
+        {
+            get => statefulInteractable;
+            set => statefulInteractable = value;
```

Comment:
@AMollis Wouldn't we want to run UpdateCollider(); when the StatefulInteractable is set as well, like we do when setting ColliderFitter?  It could already be disabled, say, and we'd need to set colliders disabled at that moment?

Reply:
Also I guess also subscribing to the newly set StateInteractable's OnEnable/OnDisable since we are supporting having it set via code as well as in Inspector?

Reply:
Oh good point @whebertML .   I'll make that change